vite-plugins/generate-build-info.js

```diff
@@ -1,5 +1,35 @@
 import { execSync } from 'child_process'
 
+/**
+ * Javascriptifaction of
+ * https://github.com/geoadmin/web-mapviewer/blob/4826f4a05df9a5a12fc3eea2a902618800fc9425/buildspec.yml#L43-L44
+ * so that it also functions properly on a window10/11 machine in PowerShell (or Window terminal)
+ *
+ * Original cmd is git show-ref | grep "$(git rev-parse HEAD)" | awk '!/refs/tags// {print $2}' |
+ * head -1
+ *
+ * @param {String} gitHash The current git hash
+ * @returns {String} The git branch without "origin/" or "refs/" portion
+ */
+function getGitBranch(gitHash) {
+    const headRefs = execSync('git show-ref --heads').toString().trim().split('\n')
+    // we now do a grep-like operation, only keeping refs that have the hash found in gitHash
+    // we also have to remove all prefixes on the branch, i.e. /refs/tags or /origin/ etc...
+    const branches = headRefs
+        .filter((ref) => ref.indexOf(gitHash) !== -1)
+        .map((matchingRef) => matchingRef.substring(matchingRef.lastIndexOf('/') + 1)) // +1 so that we skip the /
+    // only taking into account the last item (like doing a head -1)
+    return branches[0]
+}
+
+const currentHash = execSync('git rev-parse HEAD').toString().trim()
+
+// We take the version from GIT_BRANCH but if not set, then take it from git show-ref
+let gitBranch = process.env.GIT_BRANCH
+if (!gitBranch) {
+    gitBranch = getGitBranch(currentHash)
+}
```

Comment:
Would it not make more sense to keep this code inside the async handler function ?

Reply:
I think it would make more sense, as if something would fail in this piece of code it would happen in post-build, and not at startup, I'll move the code inside